starknet_py/tests/e2e/account/account_test.py

```diff
@@ -0,0 +1,164 @@
+from unittest.mock import patch, AsyncMock
+
+import pytest
+
+from starknet_py.constants import FEE_CONTRACT_ADDRESS
+from starknet_py.contract import Contract
+from starknet_py.net import KeyPair
+from starknet_py.net.account.account import Account
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.models import parse_address, StarknetChainId
+from starknet_py.net.networks import TESTNET, MAINNET
+from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
+from starknet_py.transaction_exceptions import TransactionRejectedError
+
+
+@pytest.mark.asyncio
+async def test_deploy_account_contract_and_sign_tx(map_contract):
+    k, v = 13, 4324
+    await (
+        await map_contract.functions["put"].invoke(k, v, max_fee=MAX_FEE)
+    ).wait_for_acceptance()
+    (resp,) = await map_contract.functions["get"].call(k)
+
+    assert resp == v
+
+
+@pytest.mark.run_on_devnet
+@pytest.mark.asyncio
+async def test_get_balance_throws_when_token_not_specified(account):
+    with pytest.raises(ValueError) as err:
+        await account.get_balance()
+
+    assert "Token_address must be specified when using a custom net address" in str(
+        err.value
+    )
+
+
+@pytest.mark.asyncio
+async def test_balance_when_token_specified(account, erc20_contract):
+    balance = await account.get_balance(erc20_contract.address)
+
+    assert balance == 200
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("net", (TESTNET, MAINNET))
+async def test_get_balance_default_token_address(net):
+    client = GatewayClient(net=net)
+    acc_client = Account(
+        client=client,
+        address="0x123",
+        key_pair=KeyPair(123, 456),
+        chain=StarknetChainId.TESTNET,
+    )
+
+    with patch(
+        "starknet_py.net.gateway_client.GatewayClient.call_contract",
+        AsyncMock(),
+    ) as mocked_call_contract:
+        mocked_call_contract.return_value = [0, 0]
+
+        await acc_client.get_balance()
+
+        call = mocked_call_contract.call_args
+
+    (call,) = call[0]
+
+    assert call.to_addr == parse_address(FEE_CONTRACT_ADDRESS)
```

Comment:
This is a unit test, isn't it?

Reply:
It's not using a real network, so I think you are right.